sdk/search/Azure.Search.Documents/src/Generated/SearchModelFactory.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Search.Documents.Indexes.Models;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary> Model factory for read-only models. </summary>
+    public static partial class SearchModelFactory
+    {
+
+        /// <summary> Initializes new instance of IndexDocumentsResult class. </summary>
+        /// <param name="results"> The list of status information for each document in the indexing request. </param>
+        /// <returns> A new <see cref="Models.IndexDocumentsResult"/> instance for mocking. </returns>
+        public static IndexDocumentsResult IndexDocumentsResult(IReadOnlyList<IndexingResult> results = default)
```

Comment:
@AlexanderSher, @pakrym - is there a way to not generate this method when we have an existing one that does the job (takes `IEnumerable<>` instead of `IReadOnlyList<>`) - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/src/Models/SearchModelFactory.cs#L303-L305

Reply:
Ught, the autogenerated one should've taken the IEnumerable and removed automatically. cc @chamons can you look into this?

Did I manage completely miss this?

Reply:
I have managed to remove the auto-generated method using `CodeGenSuppress[]`, thanks to suggestion from @tg-msft .
I am not sure how to get rid of the now empty class. However, the public API is not impacted anymore.

Reply:
Maybe https://github.com/azure/autorest.csharp#make-a-model-internal would hide the class?  Feels like a bug if I'm understanding correctly, but I'm fine getting a little hacky if it helps us release today.

Reply:
But we do want to keep it public here. We have used `CodeGenModel` to indicate wholesale replacement of the model factory class. That part works because autogen is not creating any methods. Now, I need a way to express that there is no need to generate an empty class.
Like I said, it does not impact any functionality or public API, so it is not blocking.